src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -80,6 +81,26 @@ public AdminCourthouse getAdminCourtHouseById(Integer id) {
         return adminCourthouse;
     }
 
+    @Override
+    public List<ExtendedCourthouse> getAllExtendedCourthouses(List<CourthouseEntity> courthouseEntities) {
```

Comment:
I think this method is more complex than it needs to be, and I see why - by using the `mapFromListEntityToListExtendedCourthouse()` method you end up with a List of ExtendedCourthouses, where each Courthouse has no association with a Region. So you then need to reconcile that in an awkard way on lines L89-99.

To me, a simpler solution would be to use the `mapFromEntityToExtendedCourthouse()` method, which would allow you to do all the mapping needed for each courthouse before moving onto the next courthouse. This is basically the same pattern as you've already used in `UserManagementServiceImpl.getUsers()`. E.g:

```java
    @Override
    public List<ExtendedCourthouse> mapFromEntitiesToExtendedCourthouses(List<CourthouseEntity> courthouseEntities) {
        List<ExtendedCourthouse> extendedCourthouses = new ArrayList<>();

        courthouseEntities
            .forEach(courthouseEntity -> {
                ExtendedCourthouse extendedCourthouse = mapper.mapFromEntityToExtendedCourthouse(courthouseEntity);
                extendedCourthouse.setRegionId(courthouseEntity.getRegion() == null ? null : courthouseEntity.getRegion().getId());
                extendedCourthouses.add(extendedCourthouse);
            });

        return extendedCourthouses;
    }
```

Reply:
updated